src/lib.rs

```diff
@@ -115,8 +115,8 @@ pub extern "C" fn cqrel_from_json(json_str: *const c_void) -> *const CResult {
 }
 
 #[no_mangle]
-pub extern "C" fn cqrel_query_json(cqrel: *const CQRel, query_str: *const c_void) -> *const c_void {
-    let cqrel: Option<&CQRel> = unsafe { (cqrel as *mut CQRel).as_ref() };
+pub unsafe extern "C" fn cqrel_query_json(cqrel: *const CQRel, query_str: *const c_void) -> *const c_void {
+    let cqrel: Option<&CQRel> = (cqrel as *mut CQRel).as_ref();
```

Comment:
For this function and the following, I was following the philosphy of "the smaller unsafe blocks the better". I don't want to mark the whole functions as unsafe because then more unsafe behavior can be added to the function without the compiler noticing.